src/iotjs.c

```diff
@@ -166,68 +177,67 @@ int iotjs_start(iotjs_environment_t* env) {
   return exit_code;
 }
 
-
-void iotjs_uv_walk_to_close_callback(uv_handle_t* handle, void* arg) {
+static void iotjs_uv_walk_to_close_callback(uv_handle_t* handle, void* arg) {
   iotjs_handlewrap_t* handle_wrap = iotjs_handlewrap_from_handle(handle);
   IOTJS_ASSERT(handle_wrap != NULL);
 
   iotjs_handlewrap_close(handle_wrap, NULL);
 }
 
-void iotjs_conf_console_out(int (*out)(int lv, const char* fmt, ...)) {
-  iotjs_set_console_out(out);
+void iotjs_dispose(iotjs_environment_t* env) {
+  // Close uv loop.
+  uv_walk(iotjs_environment_loop(env), iotjs_uv_walk_to_close_callback, NULL);
+  uv_run(iotjs_environment_loop(env), UV_RUN_DEFAULT);
+
+  int res = uv_loop_close(iotjs_environment_loop(env));
+  IOTJS_ASSERT(res == 0);
+
+  // Release builtin modules.
+  iotjs_module_list_cleanup();
+
+  // Release JerryScript engine.
+  jerry_cleanup();
 }
 
-int iotjs_entry(int argc, char** argv) {
-  int ret_code = 0;
+void iotjs_terminate(iotjs_environment_t* env) {
+  // Release environment.
+  iotjs_environment_release();
 
-  // Initialize IoT.js
+  iotjs_debuglog_release();
 
-  iotjs_debuglog_init();
+  return;
+}
+
+void iotjs_conf_console_out(int (*out)(int lv, const char* fmt, ...)) {
+  iotjs_set_console_out(out);
+}
 
+int iotjs_entry(int argc, char** argv) {
```

Comment:
Two newlines between functions please (perhaps the tizen specific files use single newlines).